website/blog/detox-20-is-out.md

```diff
@@ -0,0 +1,260 @@
+* improved integration with test runners,
+* configurable logging subsystem,
+* headless mode for iOS via configs and CLI,
+* reversing TCP ports via Android app configs,
+* and more optimizations to land in the next minor versions.
+
+## Genymotion SaaS
+
+**Highlights**: [`Using Genymotion SaaS`].
+
+Two years ago we [added elementary support][Genymotion PR] for cloud-based Android emulators provided by [Genymotion SaaS] and started a beta testing phase across mobile projects at Wix.
+
+Previously our infra engineers had been maintaining Android virtual devices on CI build agents on their own, so switching to cloud devices cleared up their time for more productive tasks. Another improvement was particularly noticeable for teams with a vast number of tests. We could reduce the duration of their CI pipelines almost by half after they scaled up from 2 parallel devices to 6.
+
+This positive impact encouraged us to adopt [Genymotion SaaS] for CI as quickly as possible, ignoring some unresolved issues in the initial pull request. For the most part, those were minor problems in global lifecycle management. Yet that made us feel uncertain about releasing it as-is, so we decided to take time and gain more production experience before taking any direction.
+
+The further experience was surprisingly smooth and rarely presented issues, spare for a few minor [glitches](https://github.com/wix/Detox/issues/3573) in advanced scenarios. Admittedly, revamping the [Detox lifecycle][`Internals API`] took us longer than expected, which is all the more reason for us to celebrate today.
+
+We're looking forward to providing our users with more opportunities for testing in the cloud, and this step is only the first of many to come. We hope you'll utilize this new feature to your delight.
+
+## Integration with test runners
+
+**Highlights**: [`Config file > Test runner`], [`Internals API`], [`Dropping Mocha support`].
+
+It took about a few months of work to formalize the contract between Detox and a test runner. While there's still a lot of place for improvement, the new Detox release refines their interaction and lays the groundwork for third-party integrations.
+
+[Mocha] was our first supported test runner, but unfortunately, it could not keep up with our scaling requirements as the number of end-to-end tests grew. [By the time][Mocha 8] it acquired the ability to run tests in parallel, we already had [to place bets][Jest PR] on another horse, and that was [Jest].
+
+We attempted to keep compatibility with both Jest and Mocha, but the farther we went, the more obvious it was that we couldn't have it both ways. As it turned out, Jest wasn't easy to get along with – our first integration with it was too simplistic. Over a couple of years of use in production, we kept discovering various issues that forced us to rewrite our "glue" code from scratch twice, and this isn't over yet. All combined didn't leave much time and energy for tinkering with Mocha anymore.
```

Comment:
Might be a good opportunity to mention your [contribution to Jest!](https://github.com/facebook/jest/pull/9397)

Reply:
I'm not that vain, 😆  and the scope of rewrites is rather broad and sprawling over years.